torch/_inductor/select_algorithm.py

```diff
@@ -1977,7 +1977,18 @@ def generate(  # type: ignore[override]
             expected_input_args,
         )
 
-        full_input_nodes = tuple(V.graph.get_buffer(k) for k in result.input_call_args)
+        # `kernel_input_nodes` are the actual inputs that will be passed to the kernel,
+        # so e.g. views of the same input are not included. `codegen_input_nodes`
+        # includes views of inputs to preserve the kernel semantics. The shape and
+        # strides of `codegen_input_nodes` will be used to infer read/writes in
+        # TemplateBuffer.extract_read_writes
+        kernel_input_nodes = tuple(
+            [V.graph.get_buffer(k) for k in result.input_call_args]
+        )
+        # Here we have (*input_nodes, *captured_buffers)
+        codegen_input_nodes = (
+            tuple(input_nodes) + kernel_input_nodes[len(expected_input_args) :]
```

Comment:
@eellison Merge failed since I did not include inputs for subgraphs and this change restores those inputs. From what I can see inputs for subgraphs don't take part in fusion so it should be fine to just use the graph inputs here?